stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/QueryChannelsController.kt

```diff
@@ -58,25 +53,19 @@ public class QueryChannelsController internal constructor(
     ) : this(filter, sort, client, domain as ChatDomainImpl)
 
     private var channelOffset = INITIAL_CHANNEL_OFFSET
-    public var newChannelEventFilter: suspend (Channel, FilterObject) -> Boolean = { channel, filterObject ->
-        client.queryChannels(
-            QueryChannelsRequest(
-                filter = Filters.and(
-                    filterObject,
-                    Filters.eq("cid", channel.cid)
-                ),
-                offset = 0,
-                limit = 1,
-                messageLimit = 0,
-                memberLimit = 0,
-            )
-        ).await()
-            .map { channels -> channels.any { it.cid == channel.cid } }
-            .let { it.isSuccess && it.data() }
-    }
 
-    public var checkFilterOnChannelUpdatedEvent: Boolean = false
     public var recoveryNeeded: Boolean = false
+    private val defaultChannelEventsHandler: DefaultChannelEventsHandler = DefaultChannelEventsHandler(client, filter)
+    public var channelEventsHandler: ChannelEventsHandler? = null
```

Comment:
Can we try to make it immutable?

Reply:
No, since it would require changes in `ChatDomain::queryChannels`. In future maybe we can iterate over it